lib/xchain/provider/provider.go

```diff
@@ -247,68 +244,83 @@ func (p *Provider) getEVMChain(chainID uint64) (netconf.Chain, ethclient.Client,
 	return chain, client, nil
 }
 
-// offsetTracker tracks the XBlockOffset for non-empty blocks.
-//
-// It supports at-least-once semantics `assignOffset` and `getOffset`
-// for monotonically incrementing heights; i.e., get and assign may
-// be called for the same or later (higher) heights multiple times,
-// but it may not be called for a previous (lower) height.
+// offsetTracker tracks and assigns the XBlockOffset.
 type offsetTracker struct {
-	mu            sync.RWMutex
-	enabled       bool
-	currentOffset uint64
-	prevHeight    *uint64
-}
-
-func newOffsetTracker(from uint64) *offsetTracker {
-	return &offsetTracker{
-		enabled:       true,
-		currentOffset: from,
+	attestInterval uint64
+	enabled        bool
```

Comment:
What is the purpose of this `enabled` field? It looks like it might be legacy -- we just set it to true on creation.

Can we just drop it?